webauthn-rs-core/src/attestation.rs

```diff
@@ -688,7 +859,337 @@ pub(crate) fn verify_apple_anonymous_attestation(
+    let att_stmt_map =
+        cbor_try_map!(att_stmt).map_err(|_| WebauthnError::AttestationStatementMapInvalid)?;
+
+    // there's only 1 version now
+    let _ver = {
+        let ver = att_stmt_map
+            .get(&serde_cbor::Value::Text("ver".to_string()))
+            .ok_or(WebauthnError::AttestationStatementVerMissing)?;
+
+        cbor_try_string!(ver).map_err(|_| WebauthnError::AttestationStatementVerInvalid)?
+    };
+
+    let response = {
+        let response = att_stmt_map
+            .get(&serde_cbor::Value::Text("response".to_string()))
+            .ok_or(WebauthnError::AttestationStatementResponseMissing)?;
+
+        cbor_try_bytes!(response).map_err(|_| WebauthnError::AttestationStatementResponseMissing)?
+    };
+
+    // Concatenate authenticatorData and clientDataHash to form the data to verify.
+    let data_to_verify: Vec<u8> = auth_data_bytes
+        .iter()
+        .chain(client_data_hash.iter())
+        .copied()
+        .collect();
+    let data_to_verify = sha256(&data_to_verify);
+
+    // 2. Verify that response is a valid SafetyNet response of version ver by following the steps indicated by the SafetyNet online documentation. As of this writing, there is only one format of the SafetyNet response and ver is reserved for future use.
+    #[derive(Debug, serde::Deserialize, serde::Serialize)]
+    #[serde(rename_all = "camelCase")]
+    struct SafteyNetAttestResponse {
+        timestamp_ms: u64,
+        apk_package_name: String,
+        apk_certificate_digest_sha256: Vec<Base64UrlSafeData>,
+        cts_profile_match: bool,
+        basic_integrity: bool,
+        evaluation_type: Option<String>,
+    }
+
+    let response_str = std::str::from_utf8(response.as_slice())
+        .map_err(|_| WebauthnError::AttestationStatementResponseInvalid)?;
+
+    #[derive(Debug, thiserror::Error)]
+    #[allow(missing_docs)]
+    enum SafetyNetError {
+        #[error("JWT error: {0}")]
+        Jwt(#[from] jwt_simple::Error),
+
+        #[error("No cert in chain")]
+        MissingCertChain,
+
+        #[error("Invalid Cert")]
+        BadCert,
+
+        #[error("Base64 error: {0}")]
+        Base64(#[from] base64::DecodeError),
+
+        #[error("openssl")]
+        OpenSSL(#[from] openssl::error::ErrorStack),
+
+        #[error("unsupported jwt alg")]
+        BadAlg,
+
+        #[error("missing nonce")]
+        MissingNonce,
+
+        #[error("nonce invalid")]
+        BadNonce,
+
+        #[error("nonce mismatch")]
+        NonceMismatch,
+
+        #[error("hostname invalid")]
+        InvalidHostname,
+
+        #[error("False CTS Profile Match")]
+        CtsProfileMatchFailed,
+
+        #[error("Timestamp too old")]
+        Expired,
+
+        #[error("Time error: {0}")]
+        Time(#[from] std::time::SystemTimeError),
+    }
+
+    use jwt_simple::prelude::*;
```

Comment:
^ as above, will need to swap to compact_jwt due to the crypto backend. We can make any changes to compact_jwt you need though.

Reply:
will need to dig into this more...